cabot/cabotapp/tests/tests_basic.py

```diff
@@ -171,6 +184,19 @@ def fake_http_404_response(*args, **kwargs):
     return resp
 
 
+def fake_tcp_success(*args, **kwargs):
+    resp = Mock()
+    resp.query.return_value = Mock()
+    return resp
+
+
+def fake_tcp_failure(*args, **kwargs):
+    resp = Mock()
+    resp.query.return_value = Mock()
+    raise socket.timeout
```

Comment:
You can use [side_effect](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect) to raise the exception